torch/csrc/api/include/torch/enum.h

```diff
@@ -26,10 +29,69 @@ const enumtype::k##name k##name; \
 }
 
 #define TORCH_ENUM_PRETTY_PRINT(name) \
-const char* operator()(enumtype::k##name& v) const { \
+const char* operator()(const enumtype::k##name& v) const { \
   return #name; \
 }
 
+// NOTE: Backstory on why we need the following two macros:
+//
+// Consider the following options class:
+//
+// ```
+// struct TORCH_API SomeOptions {
+//   typedef c10::variant<enumtype::kNone, enumtype::kMean, enumtype::kSum> reduction_t;
+//   SomeOptions(reduction_t reduction = torch::kMean) : reduction_(reduction) {}
+//
+//   TORCH_ARG(reduction_t, reduction);
+// };
+// ```
+//
+// and the functional that uses it:
+//
+// ```
+// Tensor some_functional(
+//     const Tensor& input,
+//     const SomeOptions& options = {}) {
```

Comment:
This smells wrong to me. Why don't you just take it by value? It's a very small struct.

Reply:
I updated the comment here and will do a sweep to take `Options` by value in all functionals in a follow-up PR. Although it doesn't fix the problem that `TORCH_OPTIONS_CTOR_VARIANT_ARG3`/`TORCH_OPTIONS_CTOR_VARIANT_ARG4` try to address though :(